src/lib.rs

```diff
@@ -232,6 +286,97 @@ pub fn link(obj_file: &Path, output: &Path) -> Result<(), io::Error> {
     }
 }
 
+/// Structure used to handle compiling C code to memory instead of to disk.
+///
+/// You should use `JIT::from_module` or `JIT::compile_string` to create instance of JIT.
+/// NOTE: JIT stands for 'Just In Time' compiled, the way that Java and JavaScript work.
+#[cfg(feature = "jit")]
+pub struct JIT {
+    module: Module<SimpleJITBackend>,
+}
+
+#[cfg(feature = "jit")]
+impl From<Module<SimpleJITBackend>> for JIT {
+    fn from(module: Module<SimpleJITBackend>) -> Self {
+        Self { module }
+    }
+}
+
+#[cfg(feature = "jit")]
+impl JIT {
+    /// Compile string and return JITed code.
+    pub fn from_string(
+        program: &str,
+        opt: &Opt,
+    ) -> (Result<Self, Error>, VecDeque<CompileWarning>) {
+        let module = initialize_jit_module();
+        let (result, warnings) = compile::<SimpleJITBackend>(module, program, opt);
+        let result = result.map(JIT::from);
+        (result, warnings)
+    }
+
+    /// Invoke this function before trying to get access to "new" compiled functions.
+    pub fn finalize(&mut self) {
+        self.module.finalize_definitions();
+    }
+    /// Get compiled function, if this function doesn't exist then `None` is returned, otherwise it's address returned.
+    ///
+    /// # Panics
+    /// Panics if function is not compiled (finalized). Try to invoke `finalize` before using `get_compiled_function`.
+    pub fn get_compiled_function(&mut self, name: &str) -> Option<*const u8> {
+        use cranelift_module::FuncOrDataId;
+
+        let name = self.module.get_name(name);
+        if let Some(FuncOrDataId::Func(id)) = name {
+            Some(self.module.get_finalized_function(id))
+        } else {
+            None
+        }
+    }
+    /// Get compiled static data, if this data doesn't exit then `None` is returned, otherwise it's andress and size returned.
```

Comment:
```suggestion
    /// Get compiled static data. If this data doesn't exist then `None` is returned, otherwise its address and size are returned.
```